File: src/main/java/org/code4everything/hutool/converter/CharsetConverter.java
package org.code4everything.hutool.converter;

import org.code4everything.hutool.Converter;

import java.nio.charset.Charset;

/**
 * @author pantao
 * @since 2020/10/31
 */
public class CharsetConverter implements Converter<Charset> {

    @Override
    public Charset string2Object(String string) {
        return Charset.forName(string.toUpperCase());
    }

    @Override
    public String object2String(Object object) {
        return object instanceof Charset ? ((Charset) object).name() : "";
    }
}

File: src/main/kotlin/org/code4everything/hutool/converter/CharsetConverter.kt
package org.code4everything.hutool.converter

import java.nio.charset.Charset
import org.code4everything.hutool.Converter

class CharsetConverter : Converter<Charset> {

    override fun string2Object(string: String): Charset = Charset.forName(string.uppercase())

    override fun object2String(any: Any): String = if (any is Charset) any.name() else ""
}
